app/src/main/java/com/kamron/pogoiv/OCRHelper.java

```diff
@@ -143,6 +144,36 @@ private double getPokemonLevelFromImg(Bitmap pokemonImage, int trainerLevel) {
     }
 
     /**
+     * Get the evolution cost for a pokemon, example, weedle : 12
+     * If there was no detected upgrade cost, returns -1
+     *
+     * @param pokemonImage The image of the full pokemon screen
+     * @return the evolution cost, or -1 on scan failure
+     */
+    public int getPokemonEvolutionCostFromImg(Bitmap pokemonImage) {
+        Bitmap evolutionCostImage =
+                Bitmap.createBitmap(pokemonImage, (int) (widthPixels * 0.625), (int) (heightPixels * 0.86),
+                        (int) (widthPixels * 0.2), (int) (heightPixels * 0.05));
+        String hash = "candyCost" + hashBitmap(evolutionCostImage);
+        String stringCacheEvoCandyCost = ocrCache.get(hash);
+        int result = -1;
+        if (stringCacheEvoCandyCost == null) { //if no cached result
+            //the dark color used for text in pogo is 76,112,114
+            evolutionCostImage = replaceColors(evolutionCostImage, 76, 112, 114, Color.WHITE, 40, false);
```

Comment:
If I remember the discussion right, you had to call replaceColors twice, but I don't see that here. How did you get around it?